graphix/pattern.py

```diff
@@ -1527,14 +1546,65 @@ def to_qasm3(self, filename):
 
     def copy(self) -> Pattern:
         result = self.__new__(self.__class__)
-        result.__seq = [cmd.model_copy() for cmd in self.__seq]
+        result.__seq = [cmd.model_copy(deep=True) for cmd in self.__seq]
         result.__input_nodes = self.__input_nodes.copy()
         result.__output_nodes = self.__output_nodes.copy()
         result.__n_node = self.__n_node
         result._pauli_preprocessed = self._pauli_preprocessed
         result.results = self.results.copy()
         return result
 
+    def __extract_pauli_nodes(self, leave_nodes: set[int] | None = None) -> dict[int, tuple[command.M]]:
+        """Remove and return all the Pauli measurements of the sequence (except on nodes in `leave_nodes`).
+
+        The return dictionary is indexed by node index. The returned
+        Pauli measurements are intended to be measured first
+        (typically, by Pauli presimulation). This is an internal
+        function since the sequence is left in a state that breaks the
+        pattern invariant: the Pauli measurement commands are removed
+        from the pattern but the nodes are still considered to be
+        measured (they are not output nodes and can appear in the
+        domains of other commands). These measurements are supposed to
+        be reinserted in the pattern and/or presimulated.
+
+        """
+        if leave_nodes is None:
+            leave_nodes = set()
+        pauli_nodes = {}
+        shift_domains = {}
+        index = 0
+        while index < len(self.__seq):
+            cmd = self.__seq[index]
+
+            def expand_domain(domain):
```

Comment:
Please do not omit type annotation.

Reply:
Fixed in 162b0cb (and the function is moved outside the loop).